temporal-sdk/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1307,8 +1299,11 @@ private void startNewCronRun(
             .setTaskQueue(startRequest.getTaskQueue())
             .setBackoffStartInterval(ProtobufTimeUtils.toProtoDuration(backoffInterval))
             .setRetryPolicy(startRequest.getRetryPolicy())
-            .setLastCompletionResult(lastCompletionResult)
-            .build();
+            .setLastCompletionResult(lastCompletionResult);
+    if (lastFailure != null) {
```

Comment:
It looks like we use both Optional and null for lastFailure inside the testservice. Let's make this consistent.